src/constructs/autoscaling/user-data.test.ts

```diff
@@ -15,6 +16,33 @@ describe("GuUserData", () => {
     publicSubnetIds: [""],
   });
 
+  const createAsg = (stack: GuStack, props: GuUserDataProps) => {
+    return new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      vpc,
+      userData: new GuUserData(stack, props).userData,
+      stageDependentProps: {
+        [Stage.CODE]: {
+          minimumInstances: 1,
+        },
+        [Stage.PROD]: {
+          minimumInstances: 3,
+        },
+      },
+    });
+  };
+
+  test("GuUserData.stripMargin", () => {
+    const actual = GuUserData.stripMargin`
+      |mkdir /etc/gu
+      |cat > /etc/gu/my-application.conf <<-'EOF'
+      |  include "application"
+      |EOF
+      |`;
+
```

Comment:
I'm a bit confused about how `stripMargin` works, especially the ``` stripMargin`some-string` ``` syntax as I can see that it takes a `template`, and an `args` parameter. Do we need to test when it takes two params, or is it doing this already in this test?

Reply:
Yeah its pretty confusing and magical isn't it!! i'll confess to copying it from [here](https://gist.github.com/jimschubert/06fea56a6d2a1e7fdbc2).

[Here's](https://www.typescriptlang.org/play?#code/PTAEDMCcHsFtQBYBckAcDOAuEBzAlukgHT5IICuARkQMZzABWes6NFlAppEsAAwBs4DgEMArP2H8AJgCZhARg4B2cFMo0ZAWABQO8OQB2NJHmgHQhSHlQBZYZHzmAFEg6xUAG2GvMoACpunt4cAMpIVgY46ACCkJDCAJ4ANKBEafZRvoYA1gbQAO4GANoAugCUvpZ4kaAA3jqgoHQG6NAeHEQe0DgugV6uZQ1NZq3tnd1OGeiDutqNzYSgkBzo5B5IoAC8oK7u-R3LUuQ0HE6TNDQpqPZIKXhlWwB8oEXCFylTRXigALSg8iUrjcSkQGNBqk4AESQsplADcQ2WSHIkHMy1W6yIyyCJycwAAOpAAPxOfEGMr49AAKnxAB9gDgUpCACTyGEI7QAXw5OgWG1QXFa5m2kOWwikXEhOl5Iw2sBW6GEOA4Wws4Wsdgc1QABkNaQARESQUDM2oCyBCzl6vUATWg5FA9hVwnyKzgHAAhLrZgs2h0uj15ehFcqZjoQGqrLYMhDtYb7CbamKJZBOSkEvbHctHa7WvLtSlSjMgA) an attempt to reason about it, but it doesn't help much!

Reply:
Ah, this is helpful https://www.tektutorialshub.com/typescript/typescript-tagged-templates.

> The Tagged Template is a template string, which is tagged by a function. The Tag is a function which has the ability to process the template string. We use it to Tag a template string by placing it before it. We do not use the `()` as in the function call.

Reply:
Here are the MDN docs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates.